openmls/src/treesync/node/leaf_node.rs

```diff
@@ -234,86 +323,60 @@ impl LeafNode {
         Ok(leaf_node)
     }
 
-    /// Update the `encryption_key` in this leaf node and re-signs it.
+    /// Update a leaf node.
+    ///
+    /// This function generates a new encryption key pair that is stored in the
+    /// key store and also returned.
     ///
-    /// Optionally, a new leaf node can be provided to update more values such as
-    /// the credential.
-    pub(crate) fn update_and_re_sign(
+    /// This function can be used when generating an update. In most other cases
+    /// a leaf node should be generated as part of a new [`KeyPackage`].
+    #[allow(clippy::too_many_arguments)]
```

Comment:
I believe this isn't needed, the default threshold is 7.

https://rust-lang.github.io/rust-clippy/master/index.html#/too_many_arguments